### ISSUE TITLE
Error when installing cabal

### ISSUE THREAD

**Issue by FabArd** [gentoo-haskell/gentoo-haskell#1175]
I tried to install cabal with the following command : emerge --ask dev-haskell/cabal and I received the following error : 

>>> Verifying ebuild manifests
>>> Emerging (1 of 12) dev-haskell/text-1.2.4.1_rc1::haskell
>>> Emerging (2 of 12) dev-haskell/mtl-2.2.2::haskell
>>> Emerging (3 of 12) dev-haskell/fail-4.9.0.0::haskell
>>> Emerging (4 of 12) dev-haskell/transformers-0.5.6.2::haskell
>>> Installing (2 of 12) dev-haskell/mtl-2.2.2::haskell
>>> Installing (4 of 12) dev-haskell/transformers-0.5.6.2::haskell
>>> Installing (3 of 12) dev-haskell/fail-4.9.0.0::haskell
>>> Emerging (5 of 12) dev-haskell/transformers-compat-0.6.6::haskell
>>> Installing (5 of 12) dev-haskell/transformers-compat-0.6.6::haskell
>>> Emerging (6 of 12) dev-haskell/tagged-0.8.6.1::haskell
>>> Failed to emerge dev-haskell/text-1.2.4.1_rc1, Log file:
>>>  '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/temp/build.log'
>>> Installing (6 of 12) dev-haskell/tagged-0.8.6.1::haskell
>>> Jobs: 5 of 12 complete, 1 failed                Load avg: 2.31, 1.51, 0.86

At the end of the installation message there was this text : 

* Messages for package dev-haskell/text-1.2.4.1_rc1:

 * Cabal package 'text-1.2.4.1' is shipped with 'PM:9.0.1' ('9.0.1').
 * Ebuild author forgot an entry in CABAL_CORE_LIB_GHC_PV='PM:8.10.3 PM:8.10.4 PM:9999'.
 * Found in /usr/lib64/ghc-9.0.1/package.conf.d.initial.
 * ERROR: dev-haskell/text-1.2.4.1_rc1::haskell failed (install phase):
 *   (no error message)
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called src_install
 *   environment, line 2107:  Called haskell-cabal_src_install
 *   environment, line 1590:  Called cabal_src_install
 *   environment, line  720:  Called cabal-pkg
 *   environment, line  593:  Called ghc-install-pkg '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/temp/text-1.2.4.1_rc1.conf'
 *   environment, line 1426:  Called check-for-collisions '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/temp/text-1.2.4.1_rc1.conf.d'
 *   environment, line  757:  Called die
 * The specific snippet of code:
 *               die;
 * 
 * If you need support, post the output of `emerge --info '=dev-haskell/text-1.2.4.1_rc1::haskell'`,
 * the complete build log and the output of `emerge -pqv '=dev-haskell/text-1.2.4.1_rc1::haskell'`.
 * The complete build log is located at '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/temp/build.log'.
 * The ebuild environment file is located at '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/temp/environment'.
 * Working directory: '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/work/text-1.2.4.1-rc1'
 * S: '/var/calculate/tmp/portage/dev-haskell/text-1.2.4.1_rc1/work/text-1.2.4.1-rc1'

 * GNU info directory index is up-to-date.

What must do ? 

Thanks.

**Comment by solpeth** [comment]
That doesn't seem right :) I'll look into it. I might need you for more details.

**Comment by solpeth** [comment]
Can I ask for which version of ghc you have installed? `ghc --version` in the command line should tell you.

**Comment by FabArd** [comment]
I have installed ghc version 9.0.1

**Comment by solpeth** [comment]
That makes sense. The latest supported ghc in `::haskell` is `8.10.4`. How did you install `ghc-9.0.1`?

**Comment by solpeth** [comment]
In any case, the reason I ask is this:

If you haven't done anything special, e.g. you haven't manually unmasked the newest `dev-lang/ghc`, then you shouldn't be able to access `ghc-9.0.1`. And that is an issue I'd like to look into, if this is the case.

But if you _have_ done something like unmasking `ghc-9.0.1` manually, then it makes sense that you're running into this build failure and the fix is simply to remove whatever changes you've made. I can work through this with you as best I can if this is the case.

**Comment by FabArd** [comment]
here are the last commands of my bash : 

emerge --ask =dev-lang/ghc-9.0.1

emerge app-eselect/eselect-repository
eselect repository list
eselect repository enable haskell

echo "*/*::haskell ~$(portageq envvar ARCH)" >> /etc/portage/package.accept_keywords/custom

emerge --ask dev-haskell/cabal

**Comment by solpeth** [comment]
Could you please post the output of `emerge -vp =dev-lang/ghc-9.0.1`?

**Comment by FabArd** [comment]
# emerge -vp =dev-lang/ghc-9.0.1

Local copy of remote index is up-to-date and will be used.

Local copy of remote index is up-to-date and will be used.

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R   *] dev-lang/ghc-9.0.1:0/9.0.1::gentoo  USE="gmp (-big-endian) -binary -doc -elfutils -ghcbootstrap -ghcmakebinary -numa -profile -test" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

**Comment by solpeth** [comment]
Could you please post the output of `emerge --info`?

**Comment by FabArd** [comment]
calculate /home/fabrice # emerge --info
Portage 3.0.17 (python 3.8.8-final-0, default/amd64/20/desktop, gcc-10.2.0, glibc-2.32-r7, 5.10.26-calculate x86_64)
=================================================================
System uname: Linux-5.10.26-calculate-x86_64-AMD_Phenom-tm-_II_X4_955_Processor-with-glibc2.2.5
KiB Mem:    16404964 total,  13848416 free
KiB Swap:   32809924 total,  32809924 free
Timestamp of repository haskell: Thu, 15 Apr 2021 06:35:24 +0000
Head commit of repository haskell: 8b7eb79997030cd135106c239ef46d11d783f483

Timestamp of repository gentoo: Wed, 14 Apr 2021 08:30:01 +0000
sh bash 5.0_p18
ld GNU ld (Gentoo 2.35.2 p1) 2.35.2
app-shells/bash:          5.0_p18::gentoo
dev-lang/perl:            5.30.3::gentoo
dev-lang/python:          2.7.18_p8::gentoo, 3.8.8_p1::gentoo
dev-util/cmake:           3.18.5::gentoo
dev-util/pkgconfig:       0.29.2::gentoo
sys-apps/baselayout:      2.7::gentoo
sys-apps/openrc:          0.42.1-r1::gentoo
sys-apps/sandbox:         2.20::gentoo
sys-devel/autoconf:       2.69-r4::gentoo
sys-devel/automake:       1.16.2-r1::gentoo
sys-devel/binutils:       2.35.2::gentoo
sys-devel/gcc:            10.2.0-r5::gentoo
sys-devel/gcc-config:     2.4::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.3::gentoo
sys-kernel/linux-headers: 5.10::gentoo (virtual/os-headers)
sys-libs/glibc:           2.32-r7::gentoo
Repositories:

custom
    location: /var/calculate/custom-overlay
    masters: calculate gentoo

haskell
    location: /var/db/repos/haskell
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/haskell.git
    masters: gentoo

calculate
    location: /var/db/repos/calculate
    sync-type: calculate
    sync-uri: git://github.com/calculatelinux/calculate.git
    masters: gentoo
    priority: 50

distros
    location: /var/db/repos/distros
    sync-type: calculate
    sync-uri: git://github.com/calculatelinux/distros.git
    masters: calculate gentoo
    priority: 50

gentoo
    location: /usr/portage
    sync-type: calculate
    sync-uri: git://github.com/calculatelinux/gentoo.git
    priority: 50

Installed sets: @autodeps, @custom
ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="*"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-O2 -march=x86-64 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo"
CXXFLAGS="-O2 -march=x86-64 -pipe"
DISTDIR="/var/calculate/distfiles"
EMERGE_DEFAULT_OPTS=" --autounmask --binpkg-changed-deps=n --binpkg-respect-use=y --jobs=4 --quiet-build=n  "
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles getbinpkg ipc-sandbox merge-sync multilib-strict network-sandbox parallel-fetch parallel-install pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch usersandbox usersync xattr"
FFLAGS="-O2 -pipe"
GENTOO_MIRRORS="http://mirror.yandex.ru/gentoo-distfiles http://distfiles.gentoo.org"
INSTALL_MASK="/etc/systemd /lib*/systemd /usr/lib*/systemd /etc/shorewall*/* -/etc/shorewall*/shorewall*.conf"
LANG="fr_FR.utf8"
LDFLAGS="-Wl,--as-needed"
MAKEOPTS="-j5"
PKGDIR="/var/calculate/packages/x86_64"
PORTAGE_BINHOST="http://miroir.linuxtricks.fr/calculate/grp/x86_64"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/calculate/tmp"
USE="3dnow 3dnowext X a52 aac acl acpi aes alsa amd64 amr audiofile avi avx berkdb bittorrent bluetooth box bzip2 cairo caps cdda cddb cdparanoia cgroups chm cli cpudetection cracklib crypt css cue cups dav1d dbus dftk djvu dri dv dvd dvdr dvdread ebook elogind encode exif faac fexif ffmpeg flac flash fma3 fma4 foomaticdb fortran ftp gadu gd gdbm gif gphoto2 gpm gstreamer gtk heif hunspell iconv id3tag idn ieee1394 imagemagick imap introspection ios ipod ipv6 irc jabber jpeg jpeg2k lame libglvnd libnotify libtirpc libwww lm-sensors logrotate lz4 lzo mad matroska mikmod mimap mmx mmxext mp3 mpeg mplayer multilib musepack ncurses networkmanager nfs nls nptl nsplugin ogg oggvorbis openal opencore-amr openexr opengl openmp opus oscar padlock pam pcre pdf pgo png policykit popcnt postscript ppp pulseaudio quicktime raw rdesktop readline real samba sasl scanner sdl seccomp semantic-desktop smp smtp speex spell split-usr srt sse sse2 sse3 sse4_1 sse4_2 sse4a ssl ssse3 svg taglib tcpd tga theora threads tiff truetype udev udisks unicode upnp upower usb userlocales v4l v4l2 vaapi vcd vdpau vhosts vnc vorbis vpx vulkan wavpack wayland webkit webp wimax win32codecs wmf wxwindows x264 x265 xanim xattr xcomposite xinerama xmp xop xv xvid xvmc zlib zstd" ABI_X86="64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" CAMERAS="*" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="3dnow 3dnowext mmx sse sse2 * mmxext" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="efi-64 efi-32 pc" INPUT_DEVICES="libinput evdev mutouch synaptics" KERNEL="linux" L10N="en be bg bs cs da de de-1901 el es es-AR es-CL es-ES es-MX et fi fr fr-CA hr hu it kk lt lv nl pl pt pt-BR pt-PT ro ru sk sl sr sq sv uk" LCD_DEVICES="* bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LIRC_DEVICES="devinput" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-3 php7-4" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_8" PYTHON_TARGETS="python3_8 python2_7" RUBY_TARGETS="ruby26" THEMES="CLDX" USERLAND="GNU" VIDEO_CARDS="amdgpu fbdev i915 i965 intel nouveau nv nvidia r100 r200 r300 r600 radeon qxl radeonsi vesa virtualbox via" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq proto steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, LC_ALL, LINGUAS, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, RUSTFLAGS

**Comment by solpeth** [comment]
Sorry, this might take a while. Issue is: `ghc-9.0.1` has no keywords, so it shouldn't be available on your arch. But it apparently is, in your case. What is the output of `cat /etc/portage/package.accept_keywords/*`?

**Comment by FabArd** [comment]
'# cat /etc/portage/package.accept_keywords/*
'# This file will be migrate to new installation

'# required by freecad (argument)
=media-gfx/freecad-0.19.1 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo
'# required by freecad (argument)
=dev-libs/OpenNI2-2.2_beta2 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo[python_single_target_python3_8]
'# required by freecad (argument)
=dev-python/pyside2-5.15.1 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo
'# required by freecad (argument)
=sci-libs/med-4.1.0 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo[python_single_target_python3_8]
'# required by freecad (argument)
=dev-python/pivy-0.6.5 ~amd64
'# required by dev-python/pivy-0.6.5::gentoo[quarter]
'# required by media-gfx/freecad-0.19.1::gentoo[python_single_target_python3_8]
'# required by freecad (argument)
=media-libs/quarter-1.1.0 ~amd64
'# required by dev-python/pyside2-5.15.1::gentoo
'# required by media-gfx/freecad-0.19.1::gentoo[python_single_target_python3_8]
'# required by freecad (argument)
=dev-python/shiboken2-5.15.1 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo
'# required by freecad (argument)
=sci-libs/opencascade-7.4.0-r4 ~amd64
'# required by media-gfx/freecad-0.19.1::gentoo
'# required by freecad (argument)
=sci-libs/orocos_kdl-1.4.0 ~amd64
'# required by =dev-lang/ghc-9.0.1 (argument)
=dev-lang/ghc-9.0.1 **
'# required by dev-lang/ghc-9.0.1::gentoo[-ghcbootstrap]
'# required by =dev-lang/ghc-9.0.1 (argument)
=app-admin/haskell-updater-1.3.2 ~amd64
*/*::haskell ~amd64

**Comment by solpeth** [comment]
Thank you. Please remove:
```
'# required by =dev-lang/ghc-9.0.1 (argument)
=dev-lang/ghc-9.0.1 **
```

from the respective file in `/etc/portage/package.accept_keywords/`, and your system should downgrade to `ghc-8.10.4`. `ghc-9.0.1` is not yet supported in `::haskell` or `::gentoo`.

`ghc` requires the coordination of a number of packages in order for the whole `::haskell` ecosystem to work together correctly. They are not yet all available for `ghc-9.0.1`.

Once you downgrade, this should resolve your issue with installing `Cabal`.

Closing, but please do reopen if issues persist with installing `Cabal` after doing these steps. Thanks!
